lmdeploy/serve/proxy/proxy.py

```diff
@@ -504,33 +642,123 @@ async def completions_v1(request: CompletionRequest, raw_request: Request = None
+            return node_manager.handle_unavailable_model(request.model)
+        logger.info(f'A Decode request is dispatched to {d_url}')
+
+        if not node_manager.pd_connection_pool.is_connected(p_url, d_url):
+            await node_manager.pd_connection_pool.connect(
+                PDConnectionMessage(
+                    p_url=p_url,
+                    d_url=d_url,
+                    protocol=node_manager.migration_protocol,
+                    rdma_config=node_manager.rdma_config,
+                ))
+
+        request_dict['migration_request'] = MigrationRequest(
+            protocol=node_manager.migration_protocol,
+            remote_engine_id=p_url,
+            remote_session_id=int(prefill_info['id']),
+            remote_block_ids=prefill_info['cache_block_ids'],
+            remote_token_id=prefill_info['remote_token_ids'][-1],
+        ).model_dump(mode='json')
+
+        start = node_manager.pre_call(d_url)
+        if request.stream is True:
+            response = node_manager.stream_generate(request_dict,
+                                                    d_url,
+                                                    '/v1/completions',
+                                                    prefill_url=p_url,
+                                                    remote_session_id=int(prefill_info['id']))
+            background_task = node_manager.create_background_tasks(d_url, start)
+            return StreamingResponse(response, background=background_task)
+        else:
+            response = await node_manager.generate(request_dict, d_url, '/v1/completions')
+            node_manager.post_call(d_url, start)
+            resp = JSONResponse(json.loads(response))
+            async with aiohttp.ClientSession() as session:
+                async with session.post(f'{p_url}/distserve/free_cache', json={'session_id':
+                                                                               prefill_info['id']}) as response:
+                    await response.json()
+            return resp
     else:
-        response = await node_manager.generate(request_dict, node_url, '/v1/completions')
-        node_manager.post_call(node_url, start)
-        return JSONResponse(json.loads(response))
+        raise ValueError(f'No serving strategy named {node_manager.serving_strategy}')
 
 
 def proxy(server_name: str = '0.0.0.0',
           server_port: int = 8000,
-          strategy: Literal['random', 'min_expected_latency', 'min_observed_latency'] = 'min_expected_latency',
+          serving_strategy: Literal['NonDisaggregated', 'Disaggregated'] = 'NonDisaggregated',
+          routing_strategy: Literal['random', 'min_expected_latency', 'min_observed_latency'] = 'min_expected_latency',
           api_keys: Optional[Union[List[str], str]] = None,
           ssl: bool = False,
           log_level: str = 'INFO',
           disable_cache_status: bool = False,
+          *,
+          disable_gdr: bool,
```

Comment:
Since arguments previous `disable_gdr` are keyword arguments, we'd better remove "*" and define `disable_gdr` as keyword argument